src/com/unbounce/treajure/map.clj

```diff
@@ -6,3 +6,24 @@
   (if (every? map? vals)
     (apply merge-with deep-merge vals)
     (last vals)))
+
+(defn update-if-exists
+  "Variation of clojure.core/update-in which performs no-operation if the
+  key is not present in the map. "
+  [m keys f & args]
+  (let [value (get-in m keys ::not-found)]
+    (if-not (identical? value ::not-found)
+      (assoc-in m keys (apply f value args))
+      m)))
+
+(defn map-values
+  "Create a new map of same keys and applying f to all the values.
+
+  Taken from plumatic/plumbing.
+  "
+  [m f]
```

Comment:
The main buy-in of using `map-values` over using `map` is in the cases where you already have a 1-arity function to apply to the values. This means you don't have to wrap your `f` in order to work with keys and values.

So, I would say no, because in the case where they want both the key/value in the callback, they can use `map` directly.

I can't think of how we could check the arity of a function so that we only call the 1-arity or 2-arity version. i.e. when working with transducers, you have to use `completing` [1] to "upgrade" your 1-arity function. I wouldn't want to force that onto users of this function.

[1]: https://clojuredocs.org/clojure.core/completing